### DISCUSSION TITLE
How to debug shuttle app with IDEA/CLION?

### DISCUSSION THREAD

**Discussion by EnvOut** [shuttle-hq/shuttle#1106] [Q&A]
Hi all. 
I am using Clion (IDEA) to write the code. 
When I try to run in Clion I have:
```--------------------------------------------
[HINT]: Run shuttle with `cargo shuttle run`
--------------------------------------------
```

Is there a way to **run shuttle inside IDEA with debug support?**

**Comment by oddgrd** [comment]
The output you shared suggests that your IDE tried to `cargo run` your project binary, but our `shuttle_runtime::main` macro adds some pieces to it (used to provision databases, subscribe to logs etc.) that expects certain arguments. These arguments are correctly provided when you run through our CLI using `cargo shuttle run`.

It's a good idea, though, it would be great if we could run Shuttle apps locally directly from an IDE like Clion or VSCode. I only have experience with the latter, for which I imagine it would be possible to achieve this with an extension. Perhaps something similar is possible for Clion? We'd welcome a feature request for this, as well as any suggestions on how to achieve it (or just what kind of functionality would be useful to you in your IDE).

**Reply by ekilmer** [reply]
I am also interested in this feature/answer to debug Shuttle app with an IDE.

> These arguments are correctly provided when you run through our CLI using cargo shuttle run.

What arguments are you talking about here? Is it possible or easy to replicate those arguments manually with `cargo run`? If not, do you know the equivalent (assumedly, multiple) commands to set up a shuttle runtime environment without the usage of `cargo shuttle run`?

> I imagine it would be possible to achieve this with an extension. Perhaps something similar is possible for Clion?

Yes, CLion has support for extensions, but maybe it's possible to set up the necessary series of run commands to provide the required runtime environment (equivalent to `cargo shuttle run`) whenever the IDE needs to debug or run something.

> We'd welcome a feature request for this, as well as any suggestions on how to achieve it (or just what kind of functionality would be useful to you in your IDE).

Are you saying to open an issue in this repo?

While I don't have a ton of experience with Rust (yet!), I have been enjoying the [Shuttle Launchpad series](https://www.shuttle.rs/launchpad/issues) while learning Rust, but I found it odd that I could not easily debug code from my IDE.

Unfortunately, I'm not really sure what a useful suggestion would be, but any sort of documentation for debugging a shuttle app would probably be a good start. Does this exist somewhere that I missed?
